### ISSUE TITLE
SpiderMonkey 36 is out

### ISSUE THREAD

**Issue by martinies** [jxcore/jxcore#191]
Master has SpiderMonkey 34, any plan to update 36?

**Comment by obastemur** [comment]
Good Idea. I will update right after couple of items I'm dealing with.

**Comment by lygstate** [comment]
I would suggest ESR version of SpiderMonkey.
Is that possible to directly embed jxcore into xulrunner?

**Comment by ktrzeciaknubisa** [comment]
@lygstate See https://github.com/jxcore/jxcore/issues/632 thread about xulrunner

**Comment by siriux** [comment]
Hi,

SM38 is out, and it seems to bring a large performance improvement.

Running octane v2 benchmark on my machive using latest jx scores 19683 while using jsshell version 38 I get 32552.

That's more than 50% improvement !!

Something similar with v8, newer versions (4.5.x) give more than 50% improvement on the same machine.

Do you know more or less when the engines are going to be updated?

Thanks

**Comment by siriux** [comment]
Hi again,

Actually, I think it can be even better to try to integrate version 45 (next standalone release) because it includes some changes that allow to run SpiderMonkey on iOS without almost no performance penalty.

You can see https://bugzilla.mozilla.org/show_bug.cgi?id=977805 (specially comment 26) for more details.

I've tried the nightly jsshell 45 using the option --non-writable-jitcode and there is almost no performance loss.

This could be a game changer for JS on mobile, and I think it's worth testing it now.

Thanks !

**Comment by lygstate** [comment]
Yes, I am also noticed of that
